### DISCUSSION TITLE
Explanation

### DISCUSSION THREAD

**Discussion by andylupu** [Leo-Corporation/Gavilya#250] [Q&A] [answered]
Can you explain how you did the search functionality , please .

**Comment by lpeyr** [comment, marked as answer]
The search box is a `ComboBox` with the `IsEditable` property set to `true`. A method adds in its `Items` each game of the user. To not only display the name of the game, but also its icon, I used the `SearchItem` `UserControl`. For the search function, it's the default autocompletion of the `ComboBox`. I just had to override the `ToString()` method of the `SearchItem`. I also used a custom style for the `ComboBox`.

You can check the code implementation here:
### `SearchItem`
#### XAML
https://github.com/Leo-Corporation/Gavilya/blob/ec7d8a6b7477c23724c3e45fb286b724f7575c5f/Gavilya/UserControls/SearchItem.xaml#L1-L18

#### C#
https://github.com/Leo-Corporation/Gavilya/blob/ec7d8a6b7477c23724c3e45fb286b724f7575c5f/Gavilya/UserControls/SearchItem.xaml.cs#L35-L93
### `App.xaml`
https://github.com/Leo-Corporation/Gavilya/blob/ec7d8a6b7477c23724c3e45fb286b724f7575c5f/Gavilya/App.xaml#L1143-L1420

### `MainWindow.xaml`
https://github.com/Leo-Corporation/Gavilya/blob/ec7d8a6b7477c23724c3e45fb286b724f7575c5f/Gavilya/MainWindow.xaml#L165-L173

### `GamesCardsPages.xaml.cs`
https://github.com/Leo-Corporation/Gavilya/blob/ec7d8a6b7477c23724c3e45fb286b724f7575c5f/Gavilya/Pages/GamesCardsPages.xaml.cs#L63-L68

**Reply by andylupu** [reply]
Thanks 🙏🏻
